app/controllers/projects_controller.rb

```diff
@@ -1,19 +1,49 @@
 class ProjectsController < ApplicationController
-    def index
-        @projects = Project.all
-    end
+  def index
+    @projects = Project.all
+  end
+
+  def show
+    @project = Project.find(params[:id])
+  end
+
+  def new
+    @project = Project.new
+  end
+
+  def edit
+    @project = Project.find(params[:id])
+  end
 
-    def show
-        @project = Project.find(params[:id])
+  def create
+    @project = Project.new(project_params)
+
+    if @project.save
+      redirect_to @project
+    else
+      render 'new'
     end
+  end
+
+  def update
+    @project = Project.find(params[:id])
 
-    def new
+    if @project.update(project_params)
+      redirect_to @project
+    else
+      render 'edit'
     end
+  end
+
+  def destroy
+    @project = Project.find(params[:id])
+    @project.destroy
 
-    def create
-        @project = Project.new(params[:article])
+    redirect_to projects_path
+  end
 
-        @project.save
-        redirect_to @project
+  private
+    def project_params
+      params.require(:project).permit(:name, :acres, :date_closed, :restricted_endowment, :cap_rate, :admin_rate, :total_upfront, :years_upfront, :earnings_begin)
```

Comment:
nit: Let's limit the column lengths to something reasonable! (We usually use ~100 characters or so.) So like:

```
params.require(:project).permit(:name, :acres, :date_closed,
  :restricted_endowment, :cap_rate, :admin_rate, :total_upfront,
  :years_upfront, :earnings_begin)